src/Exception/ConnectException.php

```diff
@@ -10,10 +10,17 @@
  */
 class ConnectException extends RequestException
 {
+
+    /**
+     * @param string $message
+     * @param \Psr\Http\Message\RequestInterface $request
+     * @param \Exception|\Throwable $previous
+     * @param mixed[] $handlerContext
+     */
     public function __construct(
         $message,
         RequestInterface $request,
-        \Exception $previous = null,
+        $previous = null,
```

Comment:
I could be wrong, but I think it has to be a subclass of Exception here: http://php.net/manual/en/language.exceptions.extending.php. I don't think you can can create an exception with a previous exception that is Throwable and not an instance of Exception.